cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -468,12 +468,15 @@ public void upgradeApplications(FullHttpRequest request, HttpResponder responder
           } catch (UnsupportedOperationException e) {
             String errorMessage = String.format("Application %s does not support upgrade.", appId);
             updateDetail = new ApplicationUpdateDetail(appId, new NotImplementedException(errorMessage));
+            LOG.error(errorMessage, e);
           } catch (InvalidArtifactException | NotFoundException e) {
             updateDetail = new ApplicationUpdateDetail(appId, e);
+            LOG.error("Application upgrade failed with exception", e);
```

Comment:
We don't log this is because the client can easily flood the logs by keep making request with an invalid artifact.

Reply:
Thanks Terence, removed for this.